addon/components/polaris-resource-list/bulk-actions.js

```diff
@@ -0,0 +1,535 @@
+
+  /**
+   * @private
+   */
+  smallScreenPopoverVisible: false,
+
+  /**
+   * @private
+   */
+  largeScreenPopoverVisible: false,
+
+  /**
+   * @private
+   */
+  containerWidth: 0,
+
+  /**
+   * @private
+   */
+  measuring: true,
+
+  /**
+   * @private
+   * {HTMLElement}
+   */
+  containerNode: null,
+
+  /**
+   * @private
+   * {HTMLElement}
+   */
+  largeScreenButtonsNode: null,
+
+  /**
+   * @private
+   * {HTMLElement}
+   */
+  moreActionsNode: null,
+
+  /**
+   * @private
+   */
+  bulkActionsWidth: 0,
+
+  /**
+   * @private
+   */
+  addedMoreActionsWidthForMeasuring: 0,
+
+  /**
+   * @private
+   * {Number[]}
+   */
+  promotedActionsWidths: computed(function() {
+    return [];
+  }),
+
+  promotedActionsToRender: computed(
+    'promotedActions',
+    'numberOfPromotedActionsToRender',
+    function() {
+      let {
+        promotedActions,
+        numberOfPromotedActionsToRender,
+      } = this.getProperties(
+        'promotedActions',
+        'numberOfPromotedActionsToRender'
+      );
+
+      if (promotedActions && numberOfPromotedActionsToRender > 0) {
+        return promotedActions.slice(0, numberOfPromotedActionsToRender);
+      }
+
+      return [];
+    }
+  ),
+
+  numberOfPromotedActionsToRender: computed(
+    'promotedActions.length',
+    'containerWidth',
+    'bulkActionsWidth',
+    'measuring',
+    'addedMoreActionsWidthForMeasuring',
+    'promotedActionsWidths',
+    function() {
+      let {
+        promotedActions,
+        containerWidth,
+        bulkActionsWidth,
+        measuring,
+        addedMoreActionsWidthForMeasuring,
+        promotedActionsWidths,
+      } = this.getProperties(
+        'promotedActions',
+        'containerWidth',
+        'bulkActionsWidth',
+        'measuring',
+        'addedMoreActionsWidthForMeasuring',
+        'promotedActionsWidths'
+      );
+
+      if (!promotedActions) {
+        return 0;
+      }
+
+      let promotedActionsLength = promotedActions.length;
+
+      if (containerWidth >= bulkActionsWidth || measuring) {
+        return promotedActionsLength;
+      }
+
+      let sufficientSpace = false;
+      let counter = promotedActionsLength - 1;
+      let totalWidth = 0;
+
+      while (!sufficientSpace && counter >= 0) {
+        totalWidth += promotedActionsWidths[counter];
+        let widthWithRemovedAction =
+          bulkActionsWidth - totalWidth + addedMoreActionsWidthForMeasuring;
+        if (containerWidth >= widthWithRemovedAction) {
+          sufficientSpace = true;
+        } else {
+          counter--;
+        }
+      }
+
+      return counter;
+    }
+  ),
+
+  hasActions: computed('promotedActions', 'actionsCollection', function() {
+    let { promotedActions, actionsCollection } = this.getProperties(
+      'promotedActions',
+      'actionsCollection'
+    );
+
+    return Boolean(
+      (promotedActions && promotedActions.length > 0) ||
+        (actionsCollection && actionsCollection.length > 0)
```

Comment:
Shouldn't this line be at the same indentation as the one above it?

Reply:
I grabbed this right from the react implementation where they have an indent 🤷‍♂️ I'll fix it up here tho.